### PR TITLE
Fix on push check fails

### LINKED ISSUE #250: On push checks failing
When I was adding a quick-start guide I noticed that when I pushed, checks would fail, but all the checks were on files I hadn't touched. I'm going to go through and fix the files so that they will pass the checks and actual mistakes will be easier to see.

### PR DESCRIPTION
**Pull request by mo-tgeddes**
Fixes all the on push check fails.

# Pull Request checklist

We appreciate your time and effort to improve the tutorial. Please keep in mind that lesson maintainers are volunteers and it may be some time before they can respond to your contribution.

* * *

## Before you start

- [x] Read [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md).
- [x] Create an [issue](https://github.com/ESMValGroup/tutorial/issues) to discuss your idea. This allows your contributions to be incorporated into the tutorial.

## Tasks

- [x] Give this pull request a descriptive title.
- [x] If you are contributing to existing lesson materials, please make sure the content conforms to the `Lesson development` section in [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md) and does not contain any spelling or grammatical errors.
- [x] If you are making a new episode, please make sure the content conforms to the `Lesson organization` and `Lesson formatting` sections in [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md) and does not contain any spelling or grammatical errors.
- [x] Preferably Codacy checks pass. Status can be seen below your pull request. If there is an error, click the link to find out why.
- [x] Preview changes on your machine before pushing them to GitHub by running `make serve`, alternatively `make docker-serve`. Please see the `Previewing your changes locally` section in [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md) for installation instructions.
- [x] All code instructions have been tested.

If you need help with any of the tasks above, please do not hesitate to ask by commenting in the issue or pull request.

* * *

Closes #250

### CODE REVIEW

File: _episodes/07-development-setup.md [hunk 1]
```diff
@@ -470,6 +471,6 @@ Congratulations! You are now ready to make a pull request.
 
 {% include links.md %}
 
-[token-authentication-requirements]: https://github.blog/2020-12-15-token-authentication-requirements-for-git-operations/#what-you-need-to-do-today
-[manage-environments]: https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html
-[install-from-source]: https://docs.esmvaltool.org/en/latest/quickstart/installation.html#install-from-source
+
+
+
```
Comment (ehogan) [inline _episodes/07-development-setup.md:476]: Remove these lines?

File: _episodes/08-diagnostics.md [hunk 2]
```diff
@@ -496,9 +499,7 @@ together your own diagnostics. Have a look at existing recipes and diagnostics i
...
-[recipe]: https://github.com/ESMValGroup/ESMValTool/blob/master/esmvaltool/recipes/examples/recipe_python.yml
-[diagnostic]: https://github.com/ESMValGroup/ESMValTool/blob/master/esmvaltool/diag_scripts/examples/diagnostic.py
-[interface]: https://docs.esmvaltool.org/projects/esmvalcore/en/latest/interfaces.html
-[shared]: https://docs.esmvaltool.org/en/latest/api/esmvaltool.diag_scripts.shared.html
-[netCDF]: https://unidata.github.io/netcdf4-python/
-[provenance]: https://docs.esmvaltool.org/en/latest/community/diagnostic.html?highlight=provenance#recording-provenance
+
+
+
+
```
Comment (ehogan) [inline _episodes/08-diagnostics.md:505]: Remove these lines?

File: _includes/links.md [hunk 3]
```diff
@@ -11,9 +12,15 @@
 [cran-knitr]: https://cran.r-project.org/package=knitr
 [cran-stringr]: https://cran.r-project.org/package=stringr
 [dc-lessons]: http://www.datacarpentry.org/lessons/
+[diagnostic]: https://github.com/ESMValGroup/ESMValTool/blob/master/esmvaltool/diag_scripts/examples/diagnostic.py
+[ds]: https://github.com/ESMValGroup/ESMValTool_Tutorial/blob/main/data/dataset.urls
 [email]: mailto:team@carpentries.org
 [github-importer]: https://import.github.com/
+[gitter-badge]: https://badges.gitter.im/ESMValGroup/Tutorial.svg
+[gitter]: https://gitter.im/ESMValGroup/Tutorial?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge
```
Comment (ehogan) [inline _includes/links.md:20]: These links aren't needed here (since they stayed in the `README.md`)

File: _includes/links.md [hunk 4]
```diff
@@ -28,22 +35,31 @@
...
 [styles]: https://github.com/carpentries/styles/
 [swc-lessons]: https://software-carpentry.org/lessons/
 [swc-releases]: https://github.com/swcarpentry/swc-releases
+[theatoga.nc]: http://esgf-data1.ceda.ac.uk/thredds/fileServer/esg_dataroot/cmip5/output1/MOHC/HadGEM2-ES/historical/mon/ocean/Omon/r1i1p1/v20110916/thetaoga/thetaoga_Omon_HadGEM2-ES_historical_r1i1p1_185912-200512.nc
+[token-authentication-requirements]: https://github.blog/2020-12-15-token-authentication-requirements-for-git-operations/#what-you-need-to-do-today
 [training]: https://carpentries.github.io/instructor-training/
 [workshop-repo]: {{ site.workshop_repo }}
 [yaml]: http://yaml.org/
+[zenodo-badge]: https://zenodo.org/badge/DOI/10.5281/zenodo.3974592.svg
+[zenodo]: https://doi.org/10.5281/zenodo.3974592
```
Comment (ehogan) [inline _includes/links.md:65]: These links aren't needed here (since they stayed in the `README.md`)

File: setup.md [hunk 5]
```diff
@@ -276,6 +281,6 @@ ESMValTool development team.
 
 {% include links.md %}
 
-[cmip5-search]: https://esgf-data.dkrz.de/search/cmip5-dkrz/
-[theatoga.nc]: http://esgf-data1.ceda.ac.uk/thredds/fileServer/esg_dataroot/cmip5/output1/MOHC/HadGEM2-ES/historical/mon/ocean/Omon/r1i1p1/v20110916/thetaoga/thetaoga_Omon_HadGEM2-ES_historical_r1i1p1_185912-200512.nc
-[ds]: https://github.com/ESMValGroup/ESMValTool_Tutorial/blob/main/data/dataset.urls
+
+
+
```
Comment (ehogan) [inline setup.md:286]: Remove these lines?

**Review by ehogan — CHANGES_REQUESTED** [review]
Thanks for getting the checks to pass @mo-tgeddes! 🥳 Just a few minor suggestions for you :) 

If you have confirmed the changes render ok, then I think this could be merged, but I wonder whether it would be worth defining some "Developer Guidelines" to include things like adding links to the `links.md` file and not including lots of log outputs (which are hard to maintain, assuming these logs get updated when the version of ESMValTool is updated?). Does anyone from the @ESMValGroup/userengagementteam have any comments on this PR or the idea of adding some "Developer Guidelines"?

File: _episodes/07-development-setup.md (hunk 1 above)
Comment (mo-tgeddes) [reply _episodes/07-development-setup.md]: done 487d0e0 :)

File: _episodes/08-diagnostics.md (hunk 2 above)
Comment (mo-tgeddes) [reply _episodes/08-diagnostics.md]: f73295c done

File: setup.md (hunk 5 above)
Comment (mo-tgeddes) [reply setup.md]: ee42a9a done

File: _includes/links.md (hunk 3 above)
Comment (mo-tgeddes) [reply _includes/links.md]: 5e35e30 done

File: _includes/links.md (hunk 4 above)
Comment (mo-tgeddes) [reply _includes/links.md]: 5e35e30 done.

**Review by ehogan — APPROVED** [review]
Thanks @mo-tgeddes! :)

**Review by SarahAlidoost — CHANGES_REQUESTED** [review]
@mo-tgeddes thanks for the fixes. I see that you are fixing issue #218 by splitting long lines over multiple lines. This solution works for normal text but not for lines related to codes and output in the lessons. Because it confuses users, especially those lines related to the recipe. For example, now the line 
``` 
  - {dataset: HadGEM2-ES, project: CMIP5, exp: historical, mip: Amon, ensemble: r1i1p1, start_year: 1960, end_year: 2005}
 ```
is broken into two lines:
```
  - {dataset: HadGEM2-ES, project: CMIP5, exp: historical, mip: Amon, ensemble:
 r1i1p1, start_year: 1960, end_year: 2005
```

if a user copies this line to a recipe, it won't work. Because recipe is in yaml format and it does not allow for this. Instead, I suggest changing the `MAX_LINE_LEN` from `100` to a higher value like `200` in [lesson_check.py](https://github.com/ESMValGroup/ESMValTool_Tutorial/blob/833dee957e28a76bae0637cfb7a583fb23eeb7bc/bin/lesson_check.py#L106) file to fix the action `lesson-check-all`.

**Comment by ehogan** [comment]
> @mo-tgeddes thanks for the fixes. I see that you are fixing issue #218 by splitting long lines over multiple lines. This solution works for normal text but not for lines related to codes and output in the lessons. Because it confuses users, especially those lines related to the recipe. For example, now the line
> 
> ```
>   - {dataset: HadGEM2-ES, project: CMIP5, exp: historical, mip: Amon, ensemble: r1i1p1, start_year: 1960, end_year: 2005}
> ```
> 
> is broken into two lines:
> 
> ```
>   - {dataset: HadGEM2-ES, project: CMIP5, exp: historical, mip: Amon, ensemble:
>  r1i1p1, start_year: 1960, end_year: 2005
> ```
> 
> if a user copies this line to a recipe, it won't work. Because recipe is in yaml format and it does not allow for this.

It is possible to format the recipes so that the datasets exist correctly over two lines, see, for example, the [radiation budget recipe](https://github.com/ESMValGroup/ESMValTool/blob/main/esmvaltool/recipes/recipe_radiation_budget.yml#L17). @mo-tgeddes, would it be possible for you to reformat any long lines in YAML files so that they can be copied and pasted without editing, please? :)

> Instead, I suggest changing the `MAX_LINE_LEN` from `100` to a higher value like `200` in [lesson_check.py](https://github.com/ESMValGroup/ESMValTool_Tutorial/blob/833dee957e28a76bae0637cfb7a583fb23eeb7bc/bin/lesson_check.py#L106) file to fix the action `lesson-check-all`.

I would strongly recommend not doing this. Maximum line lengths are used, for example, to help with readability of files. I would argue that setting `MAX_LINE_LEN` to `200` would result in highly unreadable files, and would, therefore, not be worth checking the maximum line length at all!

As mentioned in my comment, providing some guidance on storing links etc. would enable those editing recipes to more easily write lines less than 100 characters long, and make the markdown files more readable and easier to maintain / change in the future.

**Comment by SarahAlidoost** [comment]
> > @mo-tgeddes thanks for the fixes. I see that you are fixing issue #218 by splitting long lines over multiple lines. This solution works for normal text but not for lines related to codes and output in the lessons. Because it confuses users, especially those lines related to the recipe. For example, now the line
> > ```
> >   - {dataset: HadGEM2-ES, project: CMIP5, exp: historical, mip: Amon, ensemble: r1i1p1, start_year: 1960, end_year: 2005}
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > is broken into two lines:
> > ```
> >   - {dataset: HadGEM2-ES, project: CMIP5, exp: historical, mip: Amon, ensemble:
> >  r1i1p1, start_year: 1960, end_year: 2005
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > if a user copies this line to a recipe, it won't work. Because recipe is in yaml format and it does not allow for this.
> 
> It is possible to format the recipes so that the datasets exist correctly over two lines, see, for example, the [radiation budget recipe](https://github.com/ESMValGroup/ESMValTool/blob/main/esmvaltool/recipes/recipe_radiation_budget.yml#L17). @mo-tgeddes, would it be possible for you to reformat any long lines in YAML files so that they can be copied and pasted without editing, please? :)
> 
> > Instead, I suggest changing the `MAX_LINE_LEN` from `100` to a higher value like `200` in [lesson_check.py](https://github.com/ESMValGroup/ESMValTool_Tutorial/blob/833dee957e28a76bae0637cfb7a583fb23eeb7bc/bin/lesson_check.py#L106) file to fix the action `lesson-check-all`.
> 
> I would strongly recommend not doing this. Maximum line lengths are used, for example, to help with readability of files. I would argue that setting `MAX_LINE_LEN` to `200` would result in highly unreadable files, and would, therefore, not be worth checking the maximum line length at all!
> 
> As mentioned in my comment, providing some guidance on storing links etc. would enable those editing recipes to more easily write lines less than 100 characters long, and make the markdown files more readable and easier to maintain / change in the future.

agree :+1: . Then let's fix indentations in the recipe.

**Review by ehogan — APPROVED** [review]
Thank you for correctly indenting the datasets @mo-tgeddes! :)

**Review by remi-kazeroni — APPROVED** [review]
Thanks for your work @mo-tgeddes! It's nice to have this fixed. The changes look good to me and I only have very minor suggestions regarding the indentation. And sorry that I broke the check again after merging my PR for the episode 9... 😆 
I let @SarahAlidoost have a final look and proceed with the merge is this is all fine.

**Review by SarahAlidoost — APPROVED** [review]
@mo-tgeddes nice work! :+1: thanks.

**Pull request merged**